server_test.go

```diff
@@ -226,3 +259,86 @@ func TestErrorResponse(t *testing.T) {
 		t.Error("response should be set even on error")
 	}
 }
+
+func TestCallContextLocal(t *testing.T) {
+	h1, h2 := makeRandomNodes()
+	defer h1.Close()
+	defer h2.Close()
+	s := NewServer(h1, "rpc")
+	c := NewClientWithServer(h2, "rpc", s)
+	var arith Arith
+	arith.ctxTracker = &ctxTracker{}
+	s.Register(&arith)
+
+	// Local
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second/2)
+	defer cancel()
+	err := c.CallContext(ctx, h2.ID(), "Arith", "Sleep", 5, &struct{}{})
+	if err == nil {
+		t.Fatal("expected an error")
+	}
+
+	if !strings.Contains(err.Error(), "context") {
+		t.Error("expected a context error:", err)
+	}
+
+	time.Sleep(200 * time.Millisecond)
```

Comment:
Is this necessary?  It seems like the context will always be canceled if the call to Sleep has already returned an error.

Reply:
.. eh you're right. :)